daily_milestone.go

```diff
@@ -156,15 +155,38 @@ func getMilestones(baseURL string, token string, projectID string) ([]simpleMile
 }
 
 // CreateMilestoneData creates new milestones with title and due date
-func createMilestoneData(advance int) []simpleMilestone {
+func createMilestoneData(advance int, milestoneTime string) []simpleMilestone {
 	today := time.Now().Local()
 	list := []simpleMilestone{}
 	for i := 0; i < advance; i++ {
-		date := today.AddDate(0, 0, i).Format("2006-01-02")
-		milestone := simpleMilestone{}
-		milestone.Title = date
-		milestone.DueDate = date
-		list = append(list, milestone)
+		if milestoneTime == "daily" {
+			date := today.AddDate(0, 0, i).Format("2006-01-02")
+			milestone := simpleMilestone{}
+			milestone.Title = date
+			milestone.DueDate = date
+			list = append(list, milestone)
+		} else if milestoneTime == "weekly" {
+
+			today = today.AddDate(0, 0, 7)
+			year, week := today.ISOWeek()
+			milestoneYear := strconv.Itoa(year)
+			milestoneWeek := strconv.Itoa(week)
+			milestones := milestoneYear + "-w" + milestoneWeek
+			milestone := simpleMilestone{}
+			milestone.Title = milestones
+			milestone.DueDate = milestones
+			list = append(list, milestone)
+		} else if milestoneTime == "monthly" {
+			date := today.AddDate(0, i, 0).Format("2006-01")
+			milestone := simpleMilestone{}
+			milestone.Title = date
+			milestone.DueDate = date
```

Comment:
Same as with the weekly DueDate. This needs to be set to an actual day. Here the last day of the specific month.